ui/app/components/transit-key-actions.js

```diff
@@ -35,213 +50,142 @@ const TRANSIT_PARAMS = {
 };
-const PARAMS_FOR_ACTION = {
+
+const PROPS_TO_KEEP = {
+  encrypt: ['plaintext', 'context', 'nonce', 'key_version'],
+  decrypt: ['ciphertext', 'context', 'nonce'],
   sign: ['input', 'hash_algorithm', 'key_version', 'prehashed', 'signature_algorithm'],
   verify: ['input', 'hmac', 'signature', 'hash_algorithm', 'prehashed'],
   hmac: ['input', 'algorithm', 'key_version'],
-  encrypt: ['plaintext', 'context', 'nonce', 'key_version'],
-  decrypt: ['ciphertext', 'context', 'nonce'],
   rewrap: ['ciphertext', 'context', 'nonce', 'key_version'],
+  datakey: [],
 };
+
 const SUCCESS_MESSAGE_FOR_ACTION = {
-  sign: 'Signed your data',
+  sign: 'Signed your data.',
   // the verify action doesn't trigger a success message
-  hmac: 'Created your hash output',
-  encrypt: 'Created a wrapped token for your data',
-  decrypt: 'Decrypted the data from your token',
-  rewrap: 'Created a new token for your data',
-  datakey: 'Generated your key',
-  export: 'Exported your key',
+  hmac: 'Created your hash output.',
+  encrypt: 'Created a wrapped token for your data.',
+  decrypt: 'Decrypted the data from your token.',
+  rewrap: 'Created a new token for your data.',
+  datakey: 'Generated your key.',
+  export: 'Exported your key.',
 };
-export default Component.extend(TRANSIT_PARAMS, {
-  store: service(),
-  flashMessages: service(),
-
-  // public attrs
-  selectedAction: null,
-  key: null,
-  isModalActive: false,
-
-  onRefresh() {},
-  init() {
-    this._super(...arguments);
-    // TODO figure out why get is needed here Ember Upgrade
-    // eslint-disable-next-line ember/no-get
-    if (this.selectedAction) {
-      return;
-    }
-    // eslint-disable-next-line ember/no-get
-    set(this, 'selectedAction', get(this, 'key.supportedActions.firstObject'));
-    assert('`key` is required for `' + this.toString() + '`.', this.getModelInfo());
-  },
-
-  didReceiveAttrs() {
-    this._super(...arguments);
-    this.checkAction();
-    if (this.selectedAction === 'export') {
-      this.setExportKeyDefaults();
-    }
-  },
-
-  setExportKeyDefaults() {
-    const exportKeyType = this.key.exportKeyTypes.firstObject;
-    const exportKeyVersion = this.key.validKeyVersions.lastObject;
-    this.setProperties({
-      exportKeyType,
-      exportKeyVersion,
-    });
-  },
-
-  keyIsRSA: computed('key.type', function () {
-    const type = this.key.type;
-    return type === 'rsa-2048' || type === 'rsa-3072' || type === 'rsa-4096';
-  }),
 
-  getModelInfo() {
-    const model = this.key || this.backend;
-    if (!model) {
-      return null;
-    }
-    const backend = model.backend || model.id;
-    const id = model.id;
-
-    return {
-      backend,
-      id,
-    };
-  },
-
-  checkAction() {
-    const currentAction = this.selectedAction;
-    const oldAction = this.oldSelectedAction;
-
-    this.resetParams(oldAction, currentAction);
-    set(this, 'oldSelectedAction', currentAction);
-  },
-
-  resetParams(oldAction, action) {
```

Comment:
it looks like we changed the functionality from the original since it seems like updateProps doesn't retain the `oldAction`. Does this mean we're transferring params as we navigate away from `datakey` and `rewrap` ?

Reply:
We are transferring params. This was a hard workflow to keep and work through. The `PROPS_TO_KEEP` are the params we transfer every time the key action changes (e.g. a user changes tabs). A way to see this in action is to encrypt something, decrypt it and then click back to the encrypt tab, you'll see the decrypt in the plaintext. This was the original functionality and I wanted to preserve it. 

And in case this was more your question, if the concern is about the queryParam, every time we change the tab the queryParam changes, which triggers a re-render of the component. I can't entirely remember how it was done before but I believed we hard refreshed the component, which is now done by the changing of the arg selectedAction

Reply:
After another reread of your comment —sorry just trying to make sure I'm answering— the concern is that because I'm not keeping track of what the previous renders selectedAction like we did before the behavior has changed.

I believe we have the same functionality for dataKey, but you're right, let me double check I handle rewrap the same.

datakey. if the oldAction was datakey, then don't set the values. That's still the functionality here because dataKey is an empty array of params to keep.

Reply:
Update: I was able to add in something for unwrap and specifically the ciphertext (not the exact same approach as before but more specific now). I also—after speaking with Scott— feel better about the purpose of this function —mainly it being a UX thing and to default to not moving any props forward if we're nervous about nailing this down. I would even go so far as to say, that maybe we only ever carry over the context. Thoughts?